File: cicte-app/src/app/services/piloto.service.ts
import { Piloto } from './../models/piloto';
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable, throwError } from 'rxjs';
import { map, retry } from 'rxjs/operators';
import { environment } from './../../environments/environment';

@Injectable({
  providedIn: 'root'
})

export class PilotoService {
  API_URL = environment.API_URL;
  constructor(public http: HttpClient) {
  }
  getQuery( query: string, request: any) {
    const url = `${this.API_URL}${ query }`;
    const headers = new HttpHeaders({
      'Content-Type': 'application/json'
    });
    const body = JSON.stringify(request);
    console.log('Paramentros enviados: ' + body );
    return this.http.post(url, body, { headers: headers });
  }

  getPilotos(piloto: Piloto) {
    return this.getQuery('pilotos/getPilotos', piloto)
                  .pipe( map(data => data['piloto']));
  }
  getPilotosById(piloto: Piloto) {
    return this.getQuery('pilotos/getPilotoById', piloto)
                  .pipe( map(data => data['piloto']));
  }
  addPiloto(piloto: Piloto) {
    return this.http.post( 'pilotos/addPilotos', piloto )
                  .pipe( map(data => data));
  }

  updatePiloto(piloto: Piloto) {
    return this.http.post( 'pilotos/updatePilotos', piloto )
                    .pipe( map(data => data));
  }

  deletePiloto(piloto: Piloto) {
    return this.getQuery('pilotos/deletePiloto', piloto)
                  .pipe( map(data => data));
  }
}

File: rest-api/src/com/espe/cicte/ws/service/PilotoService.java
package com.espe.cicte.ws.service;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import com.espe.cicte.ws.db.PilotoDB;
import com.espe.cicte.ws.dto.Piloto;
import com.espe.cicte.ws.dto.PilotoRequest;
import com.espe.cicte.ws.dto.PilotoResponse;

@Path("/cicte/pilotos")
public class PilotoService {
	
	private static PilotoDB pilotoDB = new PilotoDB();

	@POST
	@Path("/addPiloto")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PilotoResponse addPiloto(Piloto request) {
		PilotoResponse response = new PilotoResponse();

		response = pilotoDB.addPiloto(request);
		
		return response;
	}
	
	@POST
	@Path("/getPilotos")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PilotoResponse getPilotos() {
		PilotoResponse response = new PilotoResponse();

		response = pilotoDB.getPilotos();
	
		return response;
	}
	
	@POST
	@Path("/getPilotoById")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PilotoResponse getPilotoById(PilotoRequest request) {
		PilotoResponse response = new PilotoResponse();

		response = pilotoDB.getPilotoById(request);
	
		return response;
	}
	
	@POST
	@Path("/getFotoPiloto")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PilotoResponse getFotoPiloto(PilotoRequest request) {
		PilotoResponse response = new PilotoResponse();

		response = pilotoDB.getFotoPiloto(request);

		return response;
	}
	
	@POST
	@Path("/updatePiloto")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PilotoResponse updatePiloto(Piloto request) {
		PilotoResponse response = new PilotoResponse();

		response = pilotoDB.updatePiloto(request);

		return response;
	}
	
	@POST
	@Path("/deletePiloto")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PilotoResponse deletePiloto(Piloto request) {
		PilotoResponse response = new PilotoResponse();

		response = pilotoDB.deletePiloto(request);

		return response;
	}
	
}
